scopehal/scopehal.cpp

```diff
@@ -409,3 +409,18 @@ string to_string_sci(double d)
 	snprintf(tmp, sizeof(tmp), "%e", d);
 	return tmp;
 }
+
+/**
+	@brief compute the next highest power of 2 of 64-bit value
+ */
+uint64_t next_pow2(uint64_t v) 
+{
+	v--;
+	v |= v >> 1;
+	v |= v >> 2;
+	v |= v >> 4;
+	v |= v >> 8;
+	v |= v >> 16;
```

Comment:
This needs an extra shift by 32 to work properly for 64bit inputs. The input `0x100000001` would fail currently, for example.

Reply:
I have done a fast test and actual version seems to work fine up to value 9223372036854775807

Reply:
I confirm your fix is good
see small tests here https://www.onlinegdb.com/mw3ZHYZsQ